src/Token/AccessToken.php

```diff
@@ -100,6 +100,21 @@ public function getUid()
     }
 
     /**
+     * Checks if the token has expired.
+     *
+     * @return boolean true if the token has expired, false otherwise.
+     * @throws InvalidArgumentException if the expires is not set on the token.
+     */
+    public function hasExpired()
+    {
+        if (!isset($this->expires)) {
+            throw new InvalidArgumentException('"expires" is not set on the token');
```

Comment:
I'm not sure that using `InvalidArgumentException` is the best choice here, because there is no argument at play. A better choice would be `RuntimeException`.